File: src/shared/classes/ExternalIds.ts
export class ExternalIds {
  isrc: string
  ean: string
  upc: string

  static from(json){
    if (json === null) return null

    let externalIds = new ExternalIds()

    externalIds.isrc = json?.isrc
    externalIds.ean = json?.ean
    externalIds.upc = json?.upc

    return externalIds
  }
}

File: src/main/java/de/wittenbude/exportify/models/embeds/ExternalIDs.java
package de.wittenbude.exportify.models.embeds;

import jakarta.persistence.Embeddable;
import lombok.Getter;
import lombok.Setter;
import lombok.experimental.Accessors;

@Getter
@Setter
@Accessors(chain = true)
@Embeddable
public class ExternalIDs {

    private String isrc;
    private String ean;
    private String upc;
}
